Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.cpp

```diff
@@ -193,7 +193,9 @@ RefPtr<PixelBuffer> RemoteImageBufferProxy::getPixelBuffer(const PixelBufferForm
         return nullptr;
     auto& mutableThis = const_cast<RemoteImageBufferProxy&>(*this);
     mutableThis.flushDrawingContextAsync();
-    auto pixelBuffer = allocator.createPixelBuffer(destinationFormat, srcRect.size());
+    IntRect sourceRectScaled = srcRect;
+    sourceRectScaled.scale(resolutionScale());
+    auto pixelBuffer = allocator.createPixelBuffer(destinationFormat, sourceRectScaled.size());
```

Comment:
This fixes the `resolutionScale()` for `getPixelBuffer()`. But I think we need to remove the assertion and address this comment in `RemoteImageBufferProxy::putPixelBuffer()`:

```
    // The math inside PixelBuffer::create() doesn't agree with the math inside ImageBufferBackend::putPixelBuffer() about how m_resolutionScale interacts with the data in the ImageBuffer.
    // This means that putPixelBuffer() is only called when resolutionScale() == 1.
    ASSERT(resolutionScale() == 1);
```

If we decide to address it in a separate patch, you may need to replace the above comment with a FIXME comment.

Reply:
Maybe we could keep the getPixelBuffer and putPixelBuffer work separate? E.g. there's nothing inherently connected between fixing a thing in getPixelBuffer to warrant FIXMEs to putPixelBuffer?
All putPixelBuffer implementations are totally broken with resolutionScale() == 1.
Based on this, I don't know what kind of FIXME should be written in which putPixelBuffer implementations.

Based on this, I'd like to merge the patch without modifying it, in order to be able to land it without re-running the bots. We could then address putPixelBuffer() issues separately?
I can make a new commit about the FIXMEs, if you prefer to have additional comments.